### PR TITLE
Reformat config machines

### PR DESCRIPTION
**Pull request by jedwards4b**
Reformats config machines so that the primary file contains only the regex field for machine name matching.   The
details for each machine are then read from files in machines/_machinename_/config_machines.xml
This is v3 of config_machines.xml, this change is backward compatible with v2.

Test suite: scripts_regression_tests.py
Test baseline:
Test namelist changes:
Test status: bit for bit

Fixes 

User interface changes?:

Update gh-pages html (Y/N)?:

### CODE REVIEW

**Review by jasonb5 — CHANGES_REQUESTED** [review]
The `config_machines_version3.xsd` schema file is missing, otherwise this looks good.

**Comment by jedwards4b** [comment]
Added the config_machines_version3.xsd and the code to pass both schemas in and check the one based on the file version.

File: CIME/XML/generic_xml.py [hunk 1]
```diff
@@ -126,8 +126,10 @@ def read(self, infile, schema=None):
             logger.debug("read: {}".format(infile))
             with open(infile, "r", encoding="utf-8") as fd:
                 self.read_fd(fd)
-
-            if schema is not None and self.get_version() > 1.0:
+            version = str(self.get_version())
+            if type(schema) is dict:
+                self.validate_xml_file(infile, schema[version])
+            elif schema is not None and self.get_version() > 1.0:
```
Comment (billsacks) [inline CIME/XML/generic_xml.py:132]: As we discussed over lunch, I am happy with this approach. But can you please add some documentation in the docstring for this function describing the possible types for schema (i.e., that it can now be a string(?) or dict, and if the latter, what the keys are)?

**Review by billsacks — CHANGES_REQUESTED** [review]
Awesome. I like the new format of storing machine information! I have just one inline request for a bit more documentation on a function you changed. Some other comments / possible requests:

- There are a few changes in here unrelated to the main changes here (cprnc and some test stuff). Would it make sense to separate these into their own PR? I don't feel strongly about this.
- Does this lead to changes in the recommended (or possibly required) way that people store stuff in their personal `.cime` directory? If so, is there somewhere where this should be documented?

File: CIME/data/config/cesm/config_files.xml [hunk 2]
```diff
@@ -43,7 +43,8 @@
     <group>case_last</group>
     <file>env_case.xml</file>
     <desc>file containing machine specifications for target model primary component (for documentation only - DO NOT EDIT)</desc>
-    <schema>$CIMEROOT/CIME/data/config/xml_schemas/config_machines.xsd</schema>
+    <schema version="2.0">$CIMEROOT/CIME/data/config/xml_schemas/config_machines.xsd</schema>
```
Comment (rljacob) [inline CIME/data/config/cesm/config_files.xml:46]: Would we need to make a similar change in E3SM's config_files.xml?

File: CIME/non_py/cprnc
Comment (rljacob) [inline CIME/non_py/cprnc:1]: Are these changes intended?

**Comment by rljacob** [comment]
This is welcome since config_machines.xml can become a giant file.  Does the code in the per-machine files look just like the XML blocks in the current combined file?

File: CIME/non_py/cprnc
Comment (jedwards4b) [reply CIME/non_py/cprnc:1]: These changes were due to a merge to master.

File: CIME/data/config/cesm/config_files.xml (hunk 2 above)
Comment (jedwards4b) [reply CIME/data/config/cesm/config_files.xml:46]: no - this change is backward compatible

**Comment by jedwards4b** [comment]
@rljacob I made it backward compatible so no changes are required in e3sm.  Yes the 
format of individual files is nearly the same as the current block - the only difference is that the NODENAME_REGEX has been moved out of  the block to the top level file.
[Here is a link](https://github.com/ESMCI/ccs_config_cesm/tree/main/machines) to the reformated cesm machines files.

File: CIME/non_py/cprnc
Comment (billsacks) [reply CIME/non_py/cprnc:1]: Weird - usually a merge to master won't show up in PR diffs like this. Is the situation different for submodules like this?

**Review by billsacks — APPROVED** [review]
Thanks for the latest change!

**Comment by rljacob** [comment]
Is there a way to do one subdir with a uniquely named file per machine instead of a uniquely named subdir for each machine?

**Comment by jedwards4b** [comment]
What is your argument for that change?   I considered it, but plan to treat config_batch.xml in the same way - so there will be a config_batch.xml file in each machine subdir.

File: CIME/non_py/cprnc
Comment (jedwards4b) [reply CIME/non_py/cprnc:1]: I agree this is a little strange and will make sure that the merge works correctly.

Comment (billsacks) [reply CIME/non_py/cprnc:1]: Great, thanks!

**Comment by rljacob** [comment]
If there's going to be more then one file in each subdir, then it makes more sense.  Does the subdir name have to match the string you pass to --machine ?

**Comment by jedwards4b** [comment]
Yes, currently the subdirectory name is expected to match the string in MACH=

**Comment by jedwards4b** [comment]
The cprnc submodule does seem to be out of sync with master, my attempts to fix the issue have failed  - @jgfouca can you have a look?  There should not be any changes in cprnc due to this PR.

**Comment by jgfouca** [comment]
@jedwards4b , I just merged https://github.com/ESMCI/cime/pull/4545 , which should put us on the latest cprnc main commit. You should rebase or upstream this branch, take the cprnc commit that's on master, and the conflict will resolve.

**Comment by jedwards4b** [comment]
@jgfouca I just did a rebase again - it still doesn't seem to have worked.

**Comment by jgfouca** [comment]
It looks like the conflict is gone?

**Comment by jgfouca** [comment]
Oh, I see this PR is still trying to change the submodule. Try this:
`git checkout origin/master $path_to_cprnc ; git add $path_to_cprnc; git commit`.  This assumes origin is the main repo, not your fork.

**Comment by jedwards4b** [comment]
Still no help:
git checkout origin/master CIME/non_py/cprnc ; git add CIME/non_py/cprnc; git commit
Updated 0 paths from b6c797123
Check Xml............................................(no files to check)Skipped
Fix End of Files.....................................(no files to check)Skipped
Trim Trailing Whitespace.............................(no files to check)Skipped
black................................................(no files to check)Skipped
pylint...............................................(no files to check)Skipped
On branch reformat_config_machines
Your branch is up to date with 'mydev/reformat_config_machines'.

Untracked files:
  (use "git add <file>..." to include in what will be committed)
	#DiagsCase.py#
	#testcase.py#
	CIME/#test_status.py#
	CIME/.#test_status.py
	CIME/ParamGen/requirements.txt
	CIME/case/#check_input_data.py#
	DiagsCase.py
	ccs_config/
	scripts/create_newcase.py.log
	scripts/create_test.log
	scripts/create_test.out
	scripts/create_test2.out
	scripts/test.out
	testcase.py

nothing added to commit but untracked files present (use "git add" to track)
derecho5: ~/sandboxes/cime
:) git push mydev
Everything up-to-date

**Comment by jgfouca** [comment]
@jedwards4b , can you push this branch to the main repo? I'd like to check it out and modify it.

**Comment by jedwards4b** [comment]
I could but I shouldn't have to, you can clone or add my fork with
git remote add

**Comment by jgfouca** [comment]
I checked out your branch and somehow your cprnc submodule is still set to an older commit. This should fix it:'

```
git fetch $esmci_remote
git cherry-pick $esmci_remote/jgfouca/jime_branch
git submodule update
git push
```

**Comment by jedwards4b** [comment]
That did it - thank you!

**Review by jgfouca — APPROVED** [review]
